addons/payment/models/payment_acquirer.py

```diff
@@ -221,6 +221,31 @@ def _check_fee_var_within_boundaries(self):
             if any(not 0 <= fee < 100 for fee in (acquirer.fees_dom_var, acquirer.fees_int_var)):
                 raise ValidationError(_("Variable fees must always be positive and below 100%."))
 
+    #=== ONCHANGE METHODS ===#
+
+    @api.onchange('state', 'allow_tokenization')
+    def _onchange_state_allow_tokenization(self):
+        tokens = self.env['payment.token'].search([('acquirer_id', 'in', self.ids)])
+        if not tokens:
+            return
+
+        tokens_sub = [token.get_linked_records_info() for token in tokens]
```

Comment:
FYI tokens_sub is `list(list(dict))`:
If you have a token A & B, linked to SUB (1 & 2) & 3, you will have `[[{}, {}], [{}]]`, I believe your `len(tokens_sub)` will always be equal to `len(tokens)` :eyes:.